DatadogSessionReplay/Sources/Recorder/ViewTreeSnapshotProducer/ViewTreeSnapshot/NodeRecorders/SwiftUI/DisplayList.swift

```diff
@@ -0,0 +1,265 @@
+            case text(StyledTextContentView, CGSize)
+            case platformView
+            case color(Color._Resolved)
+            case unknown
+        }
+
+        let seed: Seed
+        let value: Value
+    }
+
+    /// An item is an element of the displayed tree in `SwiftUI`.
+    ///
+    /// The item has a unique identity, a frame in the current branch, and a value.
+    /// A value can be:
+    /// - An effect (node): which applied an effect on a sub-tree.
+    /// - A content (leaf): The content of the displayed item.
+    internal struct Item {
+        internal enum Value {
+            case effect(Effect, DisplayList)
+            case content(Content)
+            case unknown
+        }
+
+        let identity: Identity
+        let frame: CGRect
+        let value: Value
+    }
+
+    let items: [Item]
+}
+
+@available(iOS 13.0, *)
+extension DisplayList: Reflection {
+    init(_ mirror: Mirror) throws {
+        items = try mirror.descendant(path: "items")
+    }
+}
+
+@available(iOS 13.0, *)
+extension DisplayList.Identity: Reflection {
+    init(_ mirror: Mirror) throws {
+        value = try mirror.descendant(path: "value")
+    }
+}
+
+@available(iOS 13.0, *)
+extension DisplayList.Seed: Reflection {
+    init(_ mirror: Mirror) throws {
+        value = try mirror.descendant(path: "value")
+    }
+}
+
+@available(iOS 13.0, *)
+extension DisplayList.ViewRenderer: Reflection {
+    init(_ mirror: Mirror) throws {
+        renderer = try mirror.descendant(path: "renderer")
+    }
+}
+
+@available(iOS 13.0, *)
+extension DisplayList.ViewUpdater: Reflection {
+    init(_ mirror: Mirror) throws {
+        viewCache = try mirror.descendant(path: "viewCache")
+        lastList = try mirror.descendant(path: "lastList")
+    }
+}
+
+@available(iOS 13.0, *)
+extension DisplayList.Effect: Reflection {
+    init(_ mirror: Mirror) throws {
+        if let _ = mirror.descendant("identity") {
+            self = .identify // never reached: because enum case has no associated value
+        } else if let (path, style, _) = mirror.descendant("clip") as? (SwiftUI.Path, SwiftUI.FillStyle, Any) {
+            self = .clip(path, style)
+        } else {
+            self = .unknown
+        }
+    }
+}
+
+@available(iOS 13.0, *)
+extension DisplayList.ViewUpdater.ViewCache: Reflection {
+    init(_ mirror: Mirror) throws {
+        map = try mirror.descendant(path: "map")
+    }
+}
+
+@available(iOS 13.0, *)
+extension DisplayList.ViewUpdater.ViewCache.Key: Reflection {
+    init(_ mirror: Mirror) throws {
+        id = try mirror.descendant(path: "id")
+    }
+}
+
+@available(iOS 13.0, *)
+extension DisplayList.Index.ID: Reflection {
+    init(_ mirror: Mirror) throws {
+        identity = try mirror.descendant(path: "identity")
+    }
+}
+
+@available(iOS 13.0, *)
+extension DisplayList.ViewUpdater.ViewInfo: Reflection {
+    init(_ mirror: Mirror) throws {
+        let view = try mirror.descendant(UIView.self, path: "view")
```

Comment:
Here we need to specify the type as it can't be inferred, the following are equivalent:
```swift
let view = try mirror.descendant(UIView.self, path: "view")
let view: UIView = try mirror.descendant(path: "view")
```